crates/lightbeam/Cargo.toml

```diff
@@ -11,19 +11,24 @@ keywords = ["webassembly", "wasm", "compile", "compiler", "jit"]
 edition = "2018"
 
 [dependencies]
-smallvec = "1.0.0"
+auto_enums = "0.7"
+capstone = "0.6.0"
+chain-one = "0.1"
+cranelift-codegen = { path = "../../cranelift/codegen", version = "0.59.0" }
+derive_more = "0.99"
 dynasm = "0.5.2"
 dynasmrt = "0.5.2"
-wasmparser = "0.51.2"
-memoffset = "0.5.3"
+hashers = "1.0"
+iter-enum = "0.2"
 itertools = "0.8.2"
-capstone = "0.6.0"
+memoffset = "0.5.3"
+more-asserts = "0.2.1"
+smallvec = "1.0.0"
+staticvec = "0.8"
 thiserror = "1.0.9"
-cranelift-codegen = { path = "../../cranelift/codegen", version = "0.59.0" }
-multi_mut = "0.1"
-either = "1.5"
 typemap = "0.3"
-more-asserts = "0.2.1"
+wasmparser = "0.51.2"
+wasm-reader = "0.2"
```

Comment:
Switching the wasm decoder is a very big change, and it's not clear what motivates it, or how it's connected to the rest of the PR. Could you split this part into a separate PR so that we can discuss it separately?

Reply:
So as we weren't integrated with the latest Wasmtime for a while, this PR includes some changes that were implemented separately, before the integration. This work was part of two features: firstly, implementing trap information, and secondly converting Lightbeam to use `R: Read` everywhere instead of requiring a slice. It was used for the trap information as it hands control of the reader back to Lightbeam, so that Lightbeam can act independently of the parser WRT source location and other metadata. Specifically, it makes it so that Lightbeam can get the current position without requiring `wasmparser` to implement a specific accessor for the specific property we want. Both of these features would be very difficult to include separately to the rest of the work, as the integration with master was implemented on top of them. You can check the tests for `wasm-reader`, though, as it has an extensive test suite comparing it against `wasmparser`.